ciao-controller/client.go

```diff
@@ -185,8 +239,24 @@ func (client *ssntpClient) ErrorNotify(err ssntp.Error, frame *ssntp.Frame) {
 		}
 		client.ctl.ds.DetachVolumeFailure(failure.InstanceUUID, failure.VolumeUUID, failure.Reason)
 
+	case ssntp.AssignPublicIPFailure:
+		var failure payloads.ErrorPublicIPFailure
+		err := yaml.Unmarshal(payload, &failure)
+		if err != nil {
+			glog.Warning("Error unmarshalling ErrorPublicIPFailure")
+			return
+		}
+
+		err = client.ctl.ds.UnMapExternalIP(failure.PublicIP)
+		if err != nil {
+			glog.Warning(err)
+			return
+		}
+
+		msg := fmt.Sprintf("Failed to map %s to %s: %s", failure.PublicIP, failure.InstanceUUID, failure.Reason.String())
+		client.ctl.ds.LogEvent(failure.TenantUUID, msg)
+
```

Comment:
Maybe we should log the event before we try to unmap the external IP.  If this call fails the event will not be logged.  Or is this intentional?